src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PreReleaseContactsService.cs

```diff
@@ -0,0 +1,299 @@
+                                    var prereleaseRole = await _usersAndRolesDbContext
+                                        .Roles
+                                        // TODO represent Roles with an Enum
+                                        .Where(r => r.Name == "Prerelease User")
+                                        .FirstAsync();
+
+                                    _usersAndRolesDbContext.Add(new UserInvite
+                                    {
+                                        Email = email.ToLower(),
+                                        Role = prereleaseRole,
+                                        Created = Now,
+                                        // TODO
+                                        CreatedBy = ""
+                                    });
+                                    await _usersAndRolesDbContext.SaveChangesAsync();
+                                }
+                            }
+                        }
+
+                        await SendPreReleaseInviteEmail(releaseId, email, newUser);
+                    }
+                })
+                .OnSuccess(_ => GetPreReleaseContactsForReleaseAsync(releaseId));
+        }
+
+
+        public async Task<Either<ActionResult, List<PrereleaseCandidateViewModel>>>
+            RemovePreReleaseContactFromReleaseAsync(
+                Guid releaseId, string email)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanAssignPrereleaseContactsToRelease)
+                .OnSuccessDo(async () =>
+                {
+                    var existingUserRole = await _context
+                        .UserReleaseRoles
+                        .Include(r => r.User)
+                        .Where(r =>
+                            r.ReleaseId == releaseId
+                            && r.User.Email.ToLower() == email.ToLower()
+                            && r.Role == ReleaseRole.PrereleaseViewer)
+                        .FirstOrDefaultAsync();
+
+                    if (existingUserRole != null)
+                    {
+                        _context.Remove(existingUserRole);
+                        await _context.SaveChangesAsync();
+                    }
+                    else
+                    {
+                        var existingInviteToRelease = await _context
+                            .UserReleaseInvites
+                            .Where(i =>
+                                i.ReleaseId == releaseId
+                                && i.Email.ToLower() == email.ToLower()
+                                && i.Role == ReleaseRole.PrereleaseViewer)
+                            .FirstOrDefaultAsync();
+
+                        if (existingInviteToRelease != null)
+                        {
+                            // TODO - also need to remove their overall UserInvite record if they have no more 
```

Comment:
This doesn't seem directly related to this PR but do you have tickets to cover the other TODO's?

Reply:
Created https://alpha-dfe-data.atlassian.net/browse/EES-1258 for this one, and added tickets for other TODO